### ISSUE TITLE
XE: can't cast any mad scientist spells except Spark

### ISSUE THREAD

**Issue by Adomatic** [CleverRaven/Cataclysm-DDA#65212]
### Describe the bug

Because of another issue, I can't easily see what the requirements are for mad scientist spells from XE.(https://github.com/CleverRaven/Cataclysm-DDA/issues/65211) However, if I go look at the requirements in json, they look like this:

```
  {
    "id": "spell_components_ion_gun",
    "type": "requirement",
    "qualities": [ { "id": "WRENCH", "level": 1 }, { "id": "SAW_W", "level": 1 } ],
    "components": [
      [ [ "magnetron", 1 ] ],
      [ [ "pipe", 2 ] ],
      [ [ "antenna", 3 ] ],
      [ [ "cable", 20 ] ],
      [ [ "scrap", 5 ] ],
      [ [ "duct_tape", 15 ] ]
    ]
  },
```

But none of them aside from Minor Spark are actually castable. It might be that there is some way to tell why if the window were bigger — there might be some legitimate reason — but I have no idea what it would be.

Even Vancian Bill (pen and bill) doesn't work.

### Attach save file

[Ryder-trimmed.tar.gz](https://github.com/CleverRaven/Cataclysm-DDA/files/11290735/Ryder-trimmed.tar.gz)

### Steps to reproduce

Start a game with magiclysm and XE
Learn a bunch of spells, including (say) Vancian Bill and Ion Rifle
Gather necessary ingredients and make sure you have enough mana.
Try to cast.

Result: greyed out items for the spells, when you hit return they do nothing.

### Expected behavior

Castable spells.

### Screenshots

<img width="928" alt="image" src="https://user-images.githubusercontent.com/57644033/233496854-69b811e0-51cd-40a1-976e-6670dc0ee942.png">

### Versions and configuration

- OS: MacOs
    - OS Version: Darwin Ottrageous.local 22.3.0 Darwin Kernel Version 22.3.0: Mon Jan 30 20:42:11 PST 2023; root:xnu-8792.81.3~2/RELEASE_X86_64 x86_64
- Game Version: d6ec466 [64-bit]
- Graphics Version: Tiles
- Game Language: System language []
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    Bionic Professions [package_bionic_professions],
    Blaze Industries [blazeindustries],
    Magiclysm [magiclysm],
    SpeedyDex [speedydex],
    Xedra Evolved [xedra_evolved],
    sees-player, run, crouch, bleed icons, retrodays [sees_player_retro],
    No Fungal Growth [no_fungal_growth],
    Portal Storms Ignore NPCs [personal_portal_storms],
    Stats Through Skills [StatsThroughSkills]
]

**Comment by Maleclypse** [comment]
I believe you are on stable? #63705

**Comment by Adomatic** [comment]
Yes. You should be able to see that from the 'Versions and configuration' section, yes?

That PR would be helpful for seeing the entire spell, but I don't see how it would be useful for actually allowing me to cast it. Unless there's some actual reason that I shouldn't be able to that you're aware of?

**Comment by Maleclypse** [comment]
I'm guessing that you are missing some of the items or have them nearby rather than directly on your person.   There might be a PR that allowed spell components to be within reach of the PC but I can't recall for sure.

**Comment by Adomatic** [comment]
Oh funny. So just because the component is green in the list (which indicates they're on me or nearby) might not mean that I can use them? That's... a thing.

If there is such a PR it isn't in the stable build. However, the stable build does indeed show things as green if they are nearby, but then not allow you to use them.

**Comment by Maleclypse** [comment]
Yep just double checked and current experimental it shows as green when the objects are within 4 or so tiles of the PC but I cannot cast the spell until I have all the objects in my inventory.
